src/Catalyst.Core.Modules.P2P.Discovery.Hastings.Tests/UnitTests/DiscoveryTestBuilder.cs

```diff
@@ -0,0 +1,327 @@
+
+namespace Catalyst.Core.Modules.P2P.Discovery.Hastings.Tests.UnitTests
+{
+    public sealed class DiscoveryTestBuilder : IDisposable
+    {
+        private bool _autoStart;
+        private int _burnIn;
+        private ICancellationTokenProvider _cancellationProvider;
+        private IHastingsCareTaker _careTaker;
+        private IHastingsOriginator _currentState;
+        private IDns _dnsClient;
+        private int _hasValidCandidatesCheckMillisecondsFrequency;
+        private ILogger _logger;
+        private IPeerClient _peerClient;
+        private IPeerMessageCorrelationManager _peerCorrelationManager;
+        private IPeerRepository _peerRepository;
+        private IPeerSettings _peerSettings;
+        private IScheduler _scheduler;
+        private int _timeout;
+        public IList<IPeerClientObservable> PeerClientObservables;
+
+        public void Dispose()
+        {
+            _peerClient?.Dispose();
+            _peerCorrelationManager?.Dispose();
+            _peerRepository?.Dispose();
+        }
+
+        public HastingDiscoveryTest Build()
+        {
+            return HastingDiscoveryTest.GetTestInstanceOfDiscovery(_logger,
+                _peerRepository,
+                _dnsClient,
+                _peerSettings,
+                _peerClient,
+                _peerCorrelationManager,
+                _cancellationProvider,
+                PeerClientObservables,
+                _autoStart,
+                _burnIn,
+                _currentState,
+                _careTaker,
+                _timeout,
+                _hasValidCandidatesCheckMillisecondsFrequency,
+                _scheduler);
+        }
+
+        public DiscoveryTestBuilder WithScheduler(IScheduler scheduler = null)
+        {
+            _scheduler = scheduler ?? Scheduler.Default;
+            return this;
+        }
+
+        public DiscoveryTestBuilder WithLogger(ILogger logger = default)
+        {
+            _logger = logger ?? Substitute.For<ILogger>();
+            return this;
+        }
+
+        public DiscoveryTestBuilder WithPeerRepository(IPeerRepository peerRepository = default, bool mock = false)
+        {
+            _peerRepository = peerRepository == default(IPeerRepository) && mock == false
+                ? Substitute.For<IPeerRepository>()
+                : peerRepository == null
+                    ? _peerRepository = DiscoveryHelper.MockPeerRepository()
+                    : _peerRepository = peerRepository;
+
+            return this;
+        }
+
+        public DiscoveryTestBuilder WithPeerSettings(IPeerSettings peerSettings = default)
+        {
+            _peerSettings = peerSettings ?? PeerSettingsHelper.TestPeerSettings();
+            return this;
+        }
+
+        public DiscoveryTestBuilder WithDns(IDns dnsClient = default,
+            bool mock = false,
+            IPeerSettings peerSettings = default)
+        {
+            _dnsClient = dnsClient == default(IDns) && mock == false
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)